conda_index/index/__init__.py

```diff
@@ -344,7 +344,7 @@ def _get_resolve_object(subdir, precs=None, repodata=None):
     sd = SubdirData(channel)
 
     # repodata = copy.deepcopy(repodata) # slower than json.dumps/load loop
-    repodata_copy = repodata.copy()
+    repodata_copy = json.loads(json.dumps(repodata))
     for group in ("packages", "packages.conda"):
```

Comment:
this section is unnecessary after doing the json.loads/dumps loop but it also suggests that deepcopy() wasn't the problem; this would have copied the dependency list?